public/video-ui/src/slices/s3Upload.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice,PayloadAction } from '@reduxjs/toolkit';
+
+export interface S3UploadState {
+  id: string | null;
+  progress: number;
+  total: number;
+  status: 'idle' | 'uploading' | 'post-processing' | 'complete' | 'error';
+}
+
+type S3UploadAction = PayloadAction<S3UploadState>
```

Comment:
If you follow the path of how the reducer method that uses this action (`s3UploadProgress`) is used or log out the action in the method, the `payload` is actually a number, not an `S3UploadState` object.

It's a bit hard to to follow as the  `s3UploadProgress` gets wrapped in the `progress` function in `startVideoUpload`, then passed round as a callback.

Reply:
This is the point  I mentioned in the huddle last week, that I wanted to check with you, Thanks for clarifying